metric/system/network/helpers.go

```diff
@@ -22,30 +22,52 @@ import (
 	sysinfotypes "github.com/elastic/go-sysinfo/types"
 )
 
+// MapProcNetCountersWithFilter converts the NetworkCountersInfo to a formatted mapstring,
+// and applies a filter to the resulting map. The filter should be an array key values, taken from /proc/PID/net/snmp or /proc/PID/net/netstat
+func MapProcNetCountersWithFilter(raw *sysinfotypes.NetworkCountersInfo, filter []string) mapstr.M {
+	return createMap(raw, filter)
+}
+
 // MapProcNetCounters converts the NetworkCountersInfo struct into a MapStr acceptable for sending upstream
 func MapProcNetCounters(raw *sysinfotypes.NetworkCountersInfo) mapstr.M {
+	return createMap(raw, []string{"all"})
+}
 
+func createMap(raw *sysinfotypes.NetworkCountersInfo, filter []string) mapstr.M {
 	eventByProto := mapstr.M{
-		"ip":       combineMap(raw.Netstat.IPExt, raw.SNMP.IP),
-		"tcp":      combineMap(raw.Netstat.TCPExt, raw.SNMP.TCP),
+		"ip":       combineMap(raw.Netstat.IPExt, raw.SNMP.IP, filter),
+		"tcp":      combineMap(raw.Netstat.TCPExt, raw.SNMP.TCP, filter),
 		"udp":      raw.SNMP.UDP,
 		"udp_lite": raw.SNMP.UDPLite,
-		"icmp":     combineMap(raw.SNMP.ICMPMsg, raw.SNMP.ICMP),
+		"icmp":     combineMap(raw.SNMP.ICMPMsg, raw.SNMP.ICMP, filter),
 	}
 
 	return eventByProto
 }
 
 // combineMap concatinates two given maps
-func combineMap(map1, map2 map[string]uint64) map[string]interface{} {
+func combineMap(map1, map2 map[string]uint64, filter []string) map[string]interface{} {
 	var compMap = make(map[string]interface{})
 
-	for k, v := range map1 {
-		compMap[k] = checkMaxConn(k, v)
-	}
-	for k, v := range map2 {
-		compMap[k] = checkMaxConn(k, v)
+	if len(filter) == 0 || filter[0] == "all" {
+		for k, v := range map1 {
+			compMap[k] = checkMaxConn(k, v)
+		}
+		for k, v := range map2 {
+			compMap[k] = checkMaxConn(k, v)
+		}
```

Comment:
what if 2 maps have the same key, do we check twice?

Reply:
So, these values are only being merged across a given protocol, and in my experience, the kernel won't report the same metric for a given protocol twice, so it should be fine? Not sure how we would recover from that anyway, since it would seemingly be the same metric reported in two different places?

Reply:
I just realised that `checkMaxConn` would just do a type cast, it's not really a big deal to run it twice for the same key. So, the current logic is that the second map has a priority over the first if the keys collide, which I think is fine.